src/transaction/txParser.ts

```diff
@@ -278,6 +279,20 @@ const parseMetaDataHash = (metaDataHash: any): Buffer | null => {
   return metaDataHash || null
 }
 
+const deconstructUnsignedTxDecoded = (
+  unsignedTxDecoded: _UnsignedTxDecoded | _UnsignedTxDecodedWithScriptWitnesses,
+): _UnsignedTxDecoded => {
+  if (unsignedTxDecoded.length === 2) return unsignedTxDecoded
+  if (unsignedTxDecoded.length === 3) {
+    const [txBody, scriptWitnesses, meta] = unsignedTxDecoded as _UnsignedTxDecodedWithScriptWitnesses
```

Comment:
There are several things I really dislike about the related code.

1. `meta` is not a name meaningful enough, especially since we have `metaDataHash` also --- assuming meta means some metadata or auxiliary data, I consider it a bad design to have two fields for the same information in `_UnsignedTxParsed`. A quick fix would be to at least rename `meta` --- maybe you have an idea?

2. Having a type `_TxAux` with constructor `TxAux`, in adddition to about 6 other types representing transactions, makes me cringe. I would suggest to merge it with `_UnsignedTxParsed`, plus add a field `originalTxDecoded` in `_UnsignedTxParsed`, if we cannot come up with a better solution. Maybe not in this PR, but the required changes are small.

3. Having different types for `_UnsignedTxDecoded` and `_UnsignedTxDecodedWithScriptWitnesses` seems odd too, I don't think we want to make type distinction for this, just as we do not have different types for tx with and without metadata or ttl. Leaving one entry from a tuple does not sound great as a solution for the problem of having that entry null. Definitely not general enough if you have more than one entry behaving like that (which is probably just behind the corner with Alonzo/Plutus).

Reply:
1. I renamed `meta` to `metaData`
2. I merged `TxAux` and `_UnsignedTxParsed` as you suggested, also added the `originalTxDecoded`
3. I removed the `_UnsignedTxDecodedWithScriptWitnesses`